src/main/java/LeaveGroup/LeaveGroupInteractor.java

```diff
@@ -5,18 +5,33 @@
 
 import java.util.Objects;
 
+/**
+ * The leaveGroup use case.
+ * The user can only leave groups that they are currently a member of
+ * in which they are NOT the group leader.
+ */
+
 public class LeaveGroupInteractor implements LeaveGroupInputBoundary {
 
     final LeaveGroupDsGateway dsGateway;
     final LeaveGroupOutputBoundary outputBoundary;
 
+    /**
+     * The interactor for the leaveGroup use case.
+     * @param dsGateway the data access interface
+     * @param outputBoundary the output boundary implemented by the leaveGroupPresenter
+     */
     public LeaveGroupInteractor(LeaveGroupDsGateway dsGateway,
                                 LeaveGroupOutputBoundary outputBoundary) {
 
         this.dsGateway = dsGateway;
         this.outputBoundary = outputBoundary;
     }
 
+    /**
+     * @param requestModel the requestModel for the leaveGroup use case
+     * @return a successView or failView depending on whether the user has left the group
+     */
     @Override
```

Comment:
Also I think it would be nice to have a more informative message for the user if they are the group leader. Something like "User is the group leader. Group leaders must be reassigned to another user before leaving the group."

Reply:
> Also I think it would be nice to have a more informative message for the user if they are the group leader. Something like "User is the group leader. Group leaders must be reassigned to another user before leaving the group."

Okay but do I have this specific error message in the use case or is it the Presenter's job to output a more specific error message that can be shown to the user? I wasn't sure. I thought maybe the presenter could do that, as the user never sees this interactor class and is only seeing a nicely formatted message from the presenter. But I'm not sure. I wanted to make the method concise.

Reply:
That is actually a very interesting point. I am pretty sure the messages are in the interactor (from my understanding) and you just send your results to the presenter (through the output boundary).  How would you have the presenter know what error message to present?

Reply:
Hmm, alright. I was thinking that since the Presenter is already supposed to format the given string to make it more appropriate for the screen, why can't it modify the "User left group" message into something more descriptive as well? But now that I'm typing this out I think it would make more sense just to have the correct message in the interactor to begin with. I'll change that. Thanks Hannah and Helena!